### PR TITLE
Browser friendly

### PR DESCRIPTION
**Pull request by Janther**
I'm experimenting using prettier within the browser and was surprised by all of the warnings and difficulty to run.

Then I noticed prettier also provides a [standalone](https://prettier.io/docs/en/browser.html) script for browsers.

This script mainly doesn't load plugins from the filesystem and you have to provide the plugin in a javascript object.

By requiring `prettier` we are loading also all of the filesystem's libraries into the build (probably Webpack then removes them as we don't use them) but warnings are raised and also a feeling of uneasiness.

This Pull Request uses `prettier/standalone` to load all of the core builders.
Also (and I'm very sorry for this) I had to remove `dir-to-object` from this plugin's dependencies and explicitly require each File. This could be done in a `prepublish` script but I feel like we would be adding an extra step just to avoid having an explicit list of files.

Now, this plugin is fully compatible with the browser and the node environments.

### CODE REVIEW

**Comment by Janther** [comment]
@mattiaerre redid the PR now has no red files in the corner :)

File: src/nodes/index.js [hunk 1]
```diff
@@ -1,5 +1,121 @@
...
+const UserDefinedTypeName = require('./UserDefinedTypeName.js');
+const UsingForDeclaration = require('./UsingForDeclaration.js');
+const VariableDeclaration = require('./VariableDeclaration.js');
+const VariableDeclarationStatement = require('./VariableDeclarationStatement.js');
+const WhileStatement = require('./WhileStatement.js');
 
-const nodes = dirToObject(__dirname, { canAdd: data => data.print });
-
-module.exports = nodes;
+module.exports = {
```
Comment (mattiaerre) [inline src/nodes/index.js]: it'd be nice to find a way to dynamically do this; 🤔

File: src/prettier-comments/language-js/comments.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ const {
     addDanglingComment,
     getNextNonSpaceNonCommentCharacterIndex
   }
-} = require('prettier');
-const privateUtil = require("../common/util");
+} = require('prettier/standalone');
+const privateUtil = require('../common/util');
```
Comment (mattiaerre) [inline src/prettier-comments/language-js/comments.js]: can we keep the `"` in this file as it was just copied pasted from the prettier main repo?

Comment (Janther) [reply src/prettier-comments/language-js/comments.js]: done

**Comment by mattiaerre** [comment]
hi @Janther what do you think about my last commit? also do you mind to resolve the conflicts? thanks

**Comment by mattiaerre** [comment]
there is also I failing test unfortunately

**Comment by Janther** [comment]
Rebased it and made the necessary changes :)

**Comment by mattiaerre** [comment]
oh noes ... I was still working on this 😭

**Comment by Janther** [comment]
😮

**Comment by Janther** [comment]
I just moved your script to a `scripts` folder and slightly changed it so it would receive any folder

**Comment by mattiaerre** [comment]
I've changed the way we should invoke it and also added some code coverage for it; what about creating another PR?

**Comment by Janther** [comment]
Don't worry, ;) I'll have a look after a meeting I have now.

**Comment by Janther** [comment]
also it looks like we had the same idea

**Comment by Janther** [comment]
fixed.
had to change the `new Date(Date.now()).toString()` and just do `Date.now()` because of me being in a different timezone was making the tests fail

**Comment by fvictorio** [comment]
I was thinking that it would be nice to do a Remix plugin for prettier, so a lot of :+1: to this.

**Comment by Janther** [comment]
I did a remix plugin using this branch :)
https://remix-prettier.netlify.com

**Comment by fvictorio** [comment]
Oh, nice!! We could add it to the readme when this is merged.

**Pull request merged**